File: app/models/enterprise.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import Boolean, DateTime, Integer, Numeric, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base


class Enterprise(Base):
    __tablename__ = "enterprises"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    address: Mapped[str | None] = mapped_column(Text, nullable=True)

    contact_person: Mapped[str | None] = mapped_column(String(255), nullable=True)
    contact_phone: Mapped[str | None] = mapped_column(String(50), nullable=True)

    latitude: Mapped[Decimal] = mapped_column(Numeric(10, 7), nullable=False)
    longitude: Mapped[Decimal] = mapped_column(Numeric(10, 7), nullable=False)
    allowed_radius_m: Mapped[int] = mapped_column(Integer, nullable=False, default=200)

    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        server_default=func.now(),
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    practice_assignments: Mapped[list["PracticeAssignment"]] = relationship(
        back_populates="enterprise",
        cascade="save-update, merge",
    )

    def __repr__(self) -> str:
        return f"Enterprise(id={self.id!r}, name={self.name!r})"

File: frontend/src/types/enterprise.ts
export type EnterpriseItem = {
  id: number
  name: string
  address: string | null
  contact_person: string | null
  contact_phone: string | null
  latitude: number
  longitude: number
  allowed_radius_m: number
  is_active: boolean
  created_at: string
  updated_at: string
}

export type EnterpriseCreatePayload = {
  name: string
  address?: string | null
  contact_person?: string | null
  contact_phone?: string | null
  latitude: number
  longitude: number
  allowed_radius_m: number
  is_active: boolean
}

export type EnterpriseUpdatePayload = Partial<EnterpriseCreatePayload>